localstack-core/localstack/state/schema.py

```diff
@@ -0,0 +1,229 @@
+import builtins
+import logging
+import types
+import typing
+
+from localstack.constants import VERSION
+
+LOG = logging.getLogger(__name__)
+
+TypeHint = types.GenericAlias | type
+
+INTERNAL_MODULE_PREFIXES = ["localstack", "moto"]
+"""Modules that starts with this prefix are considered internal classes and are evaluated"""
+
+
+AttributeName = str
+FQN = str
+SerializedHint = str | dict[str, typing.Any]
+
+AttributeSchema = dict[AttributeName, SerializedHint]
+"""Maps an attribute name its serialized hints"""
+
+AdditionalClasses = dict[FQN, AttributeSchema]
+"""Maps the a FQN of a class to its Attribute Schema"""
+
+TAG_TYPE = "LS/TYPE"
+TAG_ARGS = "LS/ARGS"
+"""Tags for subscribed types and their args. See ``StoreSchemaBuilder`` for examples."""
```

Comment:
I think this can be simplified. Can `TAG_TYPE` and `TAG_ARGS` be statically defined instead of constants? This will also allow defining `SerializedHint` recursively.